src/api-service/__app__/onefuzzlib/azure/creds.py

```diff
@@ -99,18 +103,68 @@ def get_regions() -> List[Region]:
     return sorted([Region(x.name) for x in locations])
 
 
-@cached
-def get_graph_client() -> GraphRbacManagementClient:
-    return GraphRbacManagementClient(get_msi(), get_subscription())
+class GraphQueryError(Exception):
+    def __init__(self, message: str, status_code: Optional[int]) -> None:
+        super(GraphQueryError, self).__init__(message)
+        self.message = message
+        self.status_code = status_code
+
+
+def query_microsoft_graph(
+    method: str,
+    resource: str,
+    params: Optional[Dict] = None,
+    body: Optional[Dict] = None,
+) -> Dict:
+    cred = get_identity()
+    access_token = cred.get_token(f"{GRAPH_RESOURCE}/.default")
+
+    url = urllib.parse.urljoin(f"{GRAPH_RESOURCE_ENDPOINT}/", resource)
+    headers = {
+        "Authorization": "Bearer %s" % access_token.token,
+        "Content-Type": "application/json",
+    }
+    response = requests.request(
+        method=method, url=url, headers=headers, params=params, json=body
+    )
+
+    if 200 <= response.status_code < 300:
+        if response.content and response.content.strip():
+            return cast(Dict, response.json())
+        else:
+            return {}
+    else:
+        error_text = str(response.content, encoding="utf-8", errors="backslashreplace")
+        raise GraphQueryError(
+            f"request did not succeed: HTTP {response.status_code} - {error_text}",
+            response.status_code,
+        )
+
+
+def query_microsoft_graph_list(
+    method: str,
+    resource: str,
+    params: Optional[Dict] = None,
+    body: Optional[Dict] = None,
+) -> List[Dict]:
+    result = query_microsoft_graph(
+        method,
+        resource,
+        params,
+        body,
+    )
+    if result["value"]:
```

Comment:
1. We don't truly _know_ that the key `"value"` must occur in `result`.
2. Even if does,  if `result["value"] == []`, then the conditional expr will evaluate `False`. IOW, as-written, I don't think it is possible for this query to return the empty list, even though that may be a valid response payload.
3. Ideally we should use an `isinstance` check before casting.

One solution could be to rewrite this as something like:
```suggestion
    value = result.get("value")
    if isinstance(value, list):
```
Here, the `isinstance` is also basically doing a `None`-check for us.